### PR TITLE
Issue/pay 182 improve api client

### PR DESCRIPTION
**Pull request by tom-mollie**
* Add API keys resource
* Add profiles->update()
* Fix incorrect @method annotations and add some new ones

### CODE REVIEW

File: examples/19-update-profile.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+     */
+    include "initialize_with_oauth.php";
+
+    /*
+     * Retrieve an existing profile by his profileId
+     */
+    $profile = $mollie->profiles->get("pfl_eA4MSz7Bvy");
+
+    /**
+     * Customer fields that can be updated.
```
Comment (tom-mollie) [inline examples/19-update-profile.php]: √

File: examples/19-update-profile.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    $profile->email   = 'info@mollie.com';
+    $profile->phone   = '0612345670';
+    $profile->categoryCode = 5399;
+    $profile = $mollie->profiles->update($profile);
+
+    echo "<p>Profile updated: " . htmlspecialchars($profile->name) . "</p>";
+}
+catch (Mollie_API_Exception $e)
+{
+    echo "API call failed: " . htmlspecialchars($e->getMessage());
```
Comment (tom-mollie) [inline examples/19-update-profile.php]: Done.

File: src/Mollie/API/Resource/Payments.php [hunk 3]
```diff
@@ -31,6 +31,7 @@
  *
  * @method Mollie_API_Object_Payment[]|Mollie_API_Object_List all($offset = 0, $limit = 0, array $filters = array())
  * @method Mollie_API_Object_Payment create(array $data, array $filters = array())
+ *
```
Comment (tom-mollie) [inline src/Mollie/API/Resource/Payments.php]: √

File: src/Mollie/API/Resource/Profiles/APIKeys.php [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+	 * @var string
+	 */
+	protected $resource_path = "profiles_apikeys";
+
+	/**
+	 * @return Mollie_API_Object_Profile_APIKey
+	 */
+	protected function getResourceObject ()
+	{
+		return new Mollie_API_Object_Profile_APIKey;
```
Comment (tom-mollie) [inline src/Mollie/API/Resource/Profiles/APIKeys.php]: This is a problem in Github's markup.

**Comment by Thijs-Riezebeek** [comment]
Looks good to me. All that's left is a minor version bump.

**Pull request merged**